lib/full-site-editing.php

```diff
@@ -28,3 +28,64 @@ function gutenberg_full_site_editing_notice() {
 	<?php
 }
 add_action( 'admin_notices', 'gutenberg_full_site_editing_notice' );
+
+/**
+ * Removes legacy pages from FSE themes.
+ */
+function gutenberg_remove_legacy_pages() {
+	if ( ! gutenberg_is_fse_theme() ) {
+		return;
+	}
+
+	global $submenu;
+	if ( isset( $submenu['themes.php'] ) ) {
+		$indexes_to_remove = array();
+		foreach ( $submenu['themes.php'] as $index => $menu_item ) {
+			if (
+				false !== strpos( $menu_item[2], 'customize.php' ) ||
+				false !== strpos( $menu_item[2], 'gutenberg-widgets' )
+			) {
+				$indexes_to_remove[] = $index;
+			}
+		}
+
+		foreach ( $indexes_to_remove as $index ) {
+			unset( $submenu['themes.php'][ $index ] );
+		}
+	}
+}
+
+add_action( 'admin_menu', 'gutenberg_remove_legacy_pages' );
+
+/**
+ * Activates the 'menu_order' filter and then hooks into 'menu_order'
+ */
+add_filter( 'custom_menu_order', '__return_true' );
+add_filter( 'menu_order', 'gutenberg_menu_order' );
+
+/**
+ * Filters WordPress default menu order
+ *
+ * @param array $menu_order Menu Order.
+ */
+function gutenberg_menu_order( $menu_order ) {
+	if ( ! gutenberg_is_fse_theme() ) {
+		return;
+	}
+
+	$new_positions = array(
+		// Position the site editor before the appearnce menu.
+		'gutenberg-edit-site' => array_search( 'themes.php', $menu_order, true ),
+	);
+
+	// traverse through the new positions and move.
+	// the items if found in the original menu_positions.
```

Comment:
Not important, just a pet peeve. 😄 
First letter should be capitalized and since both lines are part of the same sentense the 1st one shouldn't have a period at the end.

```suggestion
	// Traverse through the new positions and move
	// the items if found in the original menu_positions.
```